### PR TITLE
Variable naming docs

### PR DESCRIPTION
**Pull request by cartogram**
This PR takes a first stab at generating some documentation around naming. I'd love as much feedback as possible.

### CODE REVIEW

**Comment by cartogram** [comment]
@lemonmade @tzvipm could I get you guys to take another look? 

@lemonmade I think we can put this inside React, but will need to break this document into language-specific stuff as this repo grows. https://github.com/Shopify/web-foundation/issues/43

File: Best practices/React/Variable naming.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+```javascript
+// bad
+const SKU = product.sku;
+const SKULabel = product.sku;
+const resourceSku = resource.sku;
+
+// good
+const sku = product.sku;
+const skuLabel = 'SKU';
+const resourceSKU = resource.sku;
```
Comment (lemonmade) [inline Best practices/React/Variable naming.md]: I think this is probably wrong given how our pascal case requiring rules will make this `ResourceSku` instead

File: Best practices/React/Variable naming.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+### Sub-component Markup
+
+Use `<componentName>Markup` in camel case for storing generated partial jsx output within a render function.
+
+```javascript
+// good
+render () {
+  const someComponentMarkup = <SomeComponent />;
+	const anotherComponentMarkup = <AnotherComponent />;
```
Comment (lemonmade) [inline Best practices/React/Variable naming.md]: Indentation

File: Best practices/React/Variable naming.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+      {someComponentMarkup}
+      {anotherComponentMarkup}
+    </>
+  );
+}
+```
+
+### Booleans
+
+Omit redundant `is/can/has` prefixes and when tracking an asynchronous operation use `<operation>InProgress`.
```
Comment (lemonmade) [inline Best practices/React/Variable naming.md]: I would separate these two from one another.

File: Best practices/React/Variable naming.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+The same rule applies when used as props.
+
+```javascript
+// bad
+<Component open={isOpen} loading={isDeleting} />
+
+// good
+<Component open={open} loading={deleteInProgress} />
+```
```
Comment (lemonmade) [inline Best practices/React/Variable naming.md]: The confusion I have seen here is for functions, where IMO it is better to do the `is/has/can` prefix (`isOpen()` > `open()`)

File: Best practices/React/Variable naming.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+
+### Functions
+
+Use `handle<Event>` for methods that are called in response to events, and for props that act as event listeners use `on<Event>`.
```
Comment (lemonmade) [inline Best practices/React/Variable naming.md]: I would make sure to call out that this is specifically for props, and specifically intended to disambiguate between what is a prop of the component (`onFoo`) from things that are methods (`handleFoo`)

File: Best practices/React/Variable naming.md [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+
+```javascript
+<Component onClick={this.handleClick} />
+```
+
+More complicated components often require prefixed `Event` names in order to specify the part of the component the handler is applied. In these cases use `on<Noun><Event>` and `handle<Noun><Event>`.
+
+```javascript
+<Component
+  onTitleChange={this.handleTitleChange}
```
Comment (lemonmade) [inline Best practices/React/Variable naming.md]: I think it is also worth noting that we seem OK with naming this something else entirely, i.e., `onAction={this.submit}`

File: Best practices/React/Variable naming.md [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+  onDescriptionChange={this.handleDescriptionChange}
+/>
+```
+
+### Arrays/Sets
+
+Use plurals for state or props that store multiples.
+
+```javascript
+<Component keys={['j', 'k']} />
```
Comment (lemonmade) [inline Best practices/React/Variable naming.md]: Might help to have the counterexample here

**Comment by TheMallen** [comment]
@cartogram  can you update this or close it? It's been here way too long :)

**Comment by cartogram** [comment]
Closing for now

**Pull request closed without merging**